### ISSUE TITLE
Google Summer of Code 2025

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#5978]
Everyone interested in GSoC 2025, welcome!

Please read [the instructions to apply](https://github.com/commons-app/commons-app-documentation/blob/master/android/gsoc.md).

Thank you :-)

**Comment by neeldoshii** [comment]
I really have few ideas for GSOC'25, should we create separate issue for it or should we have discussion in this same thread?

**Comment by nicolas-raoul** [comment]
@neeldoshii 
Fantastic!
Creating GitHub issues would be the best to get feedback from several people. Or you can record them in your Phabricator draft if you prefer.
No need to post them in this ticket, I believe. :-)

**Comment by neeldoshii** [comment]
Discussion
> Fix 1 bug by following the instructions at the "Developers" section [here](https://github.com/commons-app/commons-app-documentation/blob/master/android/Volunteers-welcome!.md#developers). Fixing bugs is a good opportunity to demonstrate that you can work on the app.

I was thinking—how about we add a `Good First Issue` label for fixing Android Studio warnings, specifically for new developers? (As when we run test its seems we see so many logs of lint issues)
Once, the GSOC era will start, I believe we could attract more fresh contributors to the repo. For instance, we could add an issue related to fixing Android Studio-related problems (something like [this](https://github.com/ankidroid/Anki-Android/issues/13282)) to help improve our linting issues.

* https://github.com/ankidroid/Anki-Android/issues/13282

The main advantage of doing this is that it would create an easier onboarding path for new contributors.

What do you think everyone?

**Comment by Debopriya-codes** [comment]
Hi @nicolas-raoul and @neeldoshii,

Thank you for sharing these insights! I'm excited about contributing to this organization and being part of GSoC 2025.

I appreciate the idea of labeling issues like fixing Android Studio warnings as Good First Issues, as it can make onboarding smoother for new contributors like me. I would love to start by working on such issues and improving my understanding of the codebase.

Could you guide me on how to begin? Should I start by exploring the current lint-related issues in the repository, or would you recommend any specific steps to familiarize me with the project setup?

Looking forward to your guidance and contributing to this great initiative!

Best regards,
Debopriya

**Comment by nicolas-raoul** [comment]
Great idea @neeldoshii !

I shamelessly stole the whole issue from our AnkiDroid friends.

@Debopriya-codes Would you mind giving it a try? https://github.com/commons-app/apps-android-commons/issues/5996 If anything feels off please comment on that issue.

Thanks both! 🙂

**Comment by Debopriya-codes** [comment]
Subject: Re: Issue #5996

Hi Nicolas,

Thank you for considering me for this! I’d be happy to give it a try and
review the issue. I’ll take a closer look and provide my feedback or
suggestions if needed.

Let me know if there’s anything specific you’d like me to focus on.

Best regards,
Debopriya

**Comment by misaochan** [comment]
Best of luck to all the GSoC applicants! 🚀

**Comment by Debopriya-codes** [comment]
Thanks a lot.

**Comment by sonalyadav1** [comment]
Hi @nicolas-raoul @neeldoshii @ review my PR #6045

**Comment by Akshaykomar890** [comment]
Hello @nicolas-raoul ,

I am interested in contributing as a developer to this project and participating in GSoC 2025. Please let me know how I can get started and if there are any prerequisites or tasks I need to complete.

Looking forward to your guidance

**Comment by neeldoshii** [comment]
Hi @Akshaykomar890.

For getting started with commons I would recommend [getting started](https://github.com/commons-app/commons-app-documentation/blob/master/android/Quick-start-guide-for-Developers.md#quick-start-guide-for-developers), all the docs for android-commons you can find [here](https://github.com/commons-app/commons-app-documentation/tree/master/android). and for gsoc read [this doc](https://github.com/commons-app/commons-app-documentation/blob/master/android/gsoc.md)

**Comment by Akshaykomar890** [comment]
Thank you for the information @neeldoshii  I’ll go through the provided documentation . I’m excited to get started and contribute to the project.

**Comment by nicolas-raoul** [comment]
Hi all,

Here is the suggested proposal for GSoC 2025: [Improve the app using AI and especially embedded LLMs](https://phabricator.wikimedia.org/T383961)

_Note: You are also free to imagine your own proposal if you prefer. Or even apply with two proposals: the proposal you imagined + the suggested proposal._

**Comment by parneet-guraya** [comment]
Hi @nicolas-raoul , I can't seem to find `wikimedia-foundation` in accepted orgs for GSOC 25. Could you confirm?

Thanks!

**Comment by neeldoshii** [comment]
seems like we didn't qualified for GSOC this year.

**Comment by nicolas-raoul** [comment]
I can't believe that! 🤯😱
What a shock if it's true...
I posted at https://www.mediawiki.org/wiki/Talk:Google_Summer_of_Code/2025#Organization_not_accepted?! hopefully that's some kind of glitch?
The Wikimedia Foundation had participated in the Google Summer of Code every year since 2006...

**Comment by rohit9625** [comment]
I also got shocked last night when `organizations` announced. Let's hope for the good :)

**Comment by RitikaPahwa4444** [comment]
@nicolas-raoul any idea why the MediaWiki page doesn't list the Commons app in the idea suggestions? We were looking forward to adding AI capabilities to the app and could have passed the criteria (if that's the reason for not getting accepted).

**Comment by parneet-guraya** [comment]
>We are happy to welcome organizations new to GSoC each year. Typically, 20-30 organizations join us for the first time, and we encourage you to apply. In 2025, we're particularly excited to expand our reach in the Security and Machine Learning domains.
 https://opensource.googleblog.com/2025/01/google-summer-of-code-2025-is-here.html

BTW, @nicolas-raoul is Outreachy an option for us? not sure about the timeline though.

**Comment by nicolas-raoul** [comment]
The wiki page is for listing projects until March.

Outreachy is totally an option yes. 🙂

**Comment by RitikaPahwa4444** [comment]
> The wiki page is for listing projects until March.

Fair. But was that included in our org application? Google expects orgs to add AI/ML as one of the topic tags if the org has any such project and the ideas list mostly complete by the deadline (Feb 11). 

My point is: if AI/ML was the focus, and we had a project involving GenAI integration, then it should have been added at the top. Of course the org admins would have better idea on why we didn't get accepted this year, but I felt it was worth checking if it was a miss from our side.

**Comment by nicolas-raoul** [comment]
Yes, in any case let's remember to list our project early next year.

**Comment by parneet-guraya** [comment]
> I'm happy to co-mentor for Outreachy as well, if we intend to participate in the next Outreachy round @nicolas-raoul ?

Do you mean this project won't be in the outreachy this year?

**Comment by misaochan** [comment]
> > I'm happy to co-mentor for Outreachy as well, if we intend to participate in the next Outreachy round [@nicolas-raoul](https://github.com/nicolas-raoul) ?
> 
> Do you mean this project won't be in the outreachy this year?

If we do participate in the next Outreachy round, it would be this year (and the deadline for us to submit our project would be [4 March](https://www.mediawiki.org/wiki/Outreachy/Round_30) it seems).

**Comment by parneet-guraya** [comment]
Yes, I read the same!

**Comment by nicolas-raoul** [comment]
We should definitely try to apply before March 4. I am on a ski trip until March 4 without a computer... @misaochan would you be willing to do that? You can count me as a mentor.

**Comment by misaochan** [comment]
@nicolas-raoul Sure, I can try to do it today. Just to be clear, I should take the task at https://phabricator.wikimedia.org/T383961 and copy it over to https://phabricator.wikimedia.org/project/view/7688/ , and that should be enough to get us considered? Is there anything else that needs doing urgently?

**Comment by misaochan** [comment]
Done at https://phabricator.wikimedia.org/T387664 , please let me know if any changes are needed.

We should also probably try to come up with a better learning outcome, I was in a hurry and just reused the summary.

**Comment by rohit9625** [comment]
I guess the initial application deadline is passed already. How could we participate this time?

**Comment by parneet-guraya** [comment]
I think the one you're talking about is for contributors, not for projects i.e 4th of March.

**Comment by rohit9625** [comment]
Yes, exactly for contributors.

**Comment by parneet-guraya** [comment]
No, contributions come after but you have to be selected first.

**Comment by nicolas-raoul** [comment]
Now we know what went wrong: **Wikimedia forgot to apply**. 🤯🤦 https://m.mediawiki.org/wiki/Talk:Google_Summer_of_Code/2025#c-Syced-20250302224000-DTankersley_(WMF)-20250302134700

Knowing that, I believe Wikimedia has all chances to be accepted as an organization in the following years as long as it applies. That's a better situation to be in rather than having to worry about rejections Wikimedia can't control.

On behalf of the community in general, I would like to apologize to everyone who invested time in this project with the prospect of participating in GSoC.

We also recognize that Outreachy is not an option for many of you due to past deadline and other conditions. For those patient enough, we are **looking forward to GSoC 2026** and will make sure similar mishaps do not happen.

**Comment by nicolas-raoul** [comment]
@misaochan Thanks for figuring out the process! I made it about Nearby, I hope it is OK. Originally I had proposed AI because it is a topic that appeals a lot to the GSoC sponsor. We can keep it for GSoC 2026.

**Comment by misaochan** [comment]
@nicolas-raoul All good with me, thanks for helping with the task! :)  I made #6237 for the Outreachy discussion.
